test/gurka/test_match.cc

```diff
@@ -126,6 +126,45 @@ D--3--4--C--5--6--E)";
   }
 }
 
+TEST(MapMatch, NodeSnapFix) {
+  const std::string ascii_map = R"(
+      B-C------------1----F---------D
+  )";
+
+  // The challenge posed by this case is that the gps point "B" is exactly the
+  // lat/lon as the point "B" in the graph.
+  gurka::nodelayout layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+
+  // Another thing about this case... if I declare the ways in this manner the
+  // original code works fine. The reason, I believe, is that this creates one
+  // edge with a trivial node-to-node (along the same edge) route.
+  //  const gurka::ways ways = {
+  //      {"BCFD", {{"highway", "primary"}}}
+  //  };
+
+  // To expose the issue and prove the fix, the ways must be declared in this
+  // manner. This results in a routing solution along two edges.
+  const gurka::ways ways = {
+      {"BC", {{"highway", "primary"}}},
+      {"CF", {{"highway", "primary"}}},
+      {"FD", {{"highway", "primary"}}},
+  };
+
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/mapmatch_node_snapping");
+
+  auto result = gurka::do_action(valhalla::Options::trace_route, map, {"B", "1", "F"}, "auto",
+                                 {{"/trace_options/search_radius", "50"}}, {}, nullptr, "via");
```

Comment:
nit: this confused me. the grid size in gurka is only 10 meters. by setting the search radius to 50 i would expect to see at least 2 candidates for your first input point `B` since `C` is also within 50 meters of `B`

Reply:
Honestly I don't remember why I added the 50 m search radius, I'll take another look.

Reply:
Yea I removed the 50 m radius, that didn't make sense.